crypto/txscript/src/lib.rs

```diff
@@ -559,6 +542,62 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_check_error_condition() {
+        let test_cases = vec![
+            ScriptTestCase {
+                script: b"\x51", // opcodes::codes::OpTrue{data: ""}
+                expected_result: Ok(()),
+            },
+            ScriptTestCase {
+                script: b"\x61", // opcodes::codes::OpNop{data: ""}
+                expected_result: Err(TxScriptError::EmptyStack),
+            },
+            ScriptTestCase {
+                script: b"\x51\x51", // opcodes::codes::OpTrue, opcodes::codes::OpTrue,
+                expected_result: Err(TxScriptError::CleanStack(1)),
+            },
+            ScriptTestCase {
+                script: b"\x00", // opcodes::codes::OpFalse{data: ""},
+                expected_result: Err(TxScriptError::EvalFalse),
+            },
+        ];
+
+        test_script_cases(test_cases)
+    }
+
+    #[test]
+    fn test_check_opif() {
+        let test_cases = vec![
+            ScriptTestCase {
+                script: b"\x63", // opcodes::codes::OpIf{data: ""}
+                expected_result: Err(TxScriptError::EmptyStack),
+            },
+            ScriptTestCase {
+                script: b"\x52\x63", // opcodes::codes::Op2{data: ""}, opcodes::codes::OpIf{data: ""}
+                expected_result: Err(TxScriptError::InvalidState("expected boolean".to_string())),
+            },
+            ScriptTestCase {
+                script: b"\x51\x63", // opcodes::codes::OpTrue{data: ""}, opcodes::codes::OpIf{data: ""}
+                expected_result: Err(TxScriptError::ErrUnbalancedConditional),
+            },
+            ScriptTestCase {
+                script: b"\x00\x63", // opcodes::codes::OpFalse{data: ""}, opcodes::codes::OpIf{data: ""}
+                expected_result: Err(TxScriptError::ErrUnbalancedConditional),
+            },
+            ScriptTestCase {
+                script: b"\x51\x63\x51\x68", // opcodes::codes::OpTrue{data: ""}, opcodes::codes::OpIf{data: ""}
+                expected_result: Ok(()),
+            },
+            ScriptTestCase {
+                script: b"\x00\x63\x51\x68", // opcodes::codes::OpTrue{data: ""}, opcodes::codes::OpIf{data: ""}
```

Comment:
Shouldn't it be `OpFalse OpIf OpTrue OpEndIf`?
And btw I think it's much more readable to write in the above way

(Some other comments seems wrong too, please check them)

Reply:
Fixed